packages/serverless/src/integrations/aws_lambda.ts

```diff
@@ -0,0 +1,235 @@
+    const setupOnceContext = this;
+
+    /** rapid runtime instance */
+    const rapidRuntime = lambdaBootstrap?.children[0].exports;
+
+    /** handler that is invoked in case of unhandled and handled exception */
+    const originalPostInvocationError = rapidRuntime.prototype.postInvocationError;
+
+    this.getCurrentHub = getCurrentHub;
+
+    const hub = this.getCurrentHub && this.getCurrentHub();
+
+    /** configured time */
+    this.configuredTimeInMilliseconds = this.awsContext.getRemainingTimeInMillis();
+
+    const timeoutWarningBuffer: number = 1500;
+    const millisToSeconds: number = 1000;
+
+    /** checking if configured Time In Milliseconds is greater than timeout Warning Buffer */
+    if (this.configuredTimeInMilliseconds > timeoutWarningBuffer) {
+      const configuredTimeInSec = Math.floor(this.configuredTimeInMilliseconds / millisToSeconds);
+      const configuredTimeInMilli = Math.floor(configuredTimeInSec * millisToSeconds);
+
+      /** check timeout flag */
+      if (this.timeoutWarning === true) {
+        this.awsContext.callbackWaitsForEmptyEventLoop = false;
+        timeOutError(configuredTimeInMilli);
+      }
+    }
+
+    /**
+     * This function is invoked when there is timeout error
+     * Here, we make sure the error has been captured by Sentry Dashboard
+     * and then re-raise the exception
+     * @param configuredTime  - configured time in seconds
+     * @hidden
+     */
+    function timeOutError(configuredTime: number): void {
+      setTimeout(() => {
+        /**
+         * setting parameters in scope which will be displayed as additional data in Sentry dashboard
+         */
+        setParameters();
+
+        const convertTimeToSeconds = Math.floor(configuredTime / millisToSeconds);
+
+        const error = new Error(
+          `WARNING : Function is expected to get timed out. Configured timeout duration = ${
+            convertTimeToSeconds + 1
+          } seconds.`,
+        );
+
+        /** capturing the exception and re-directing it to the Sentry Dashboard */
+        hub.captureException(error);
+        Sentry.flush(2000);
+      }, configuredTime);
+    }
+
+    /**
+     * This function sets Additional Runtime Data which are displayed in Sentry Dashboard
+     * @param scope  - holds additional event information
+     * @hidden
+     */
+    function setAdditionalRuntimeData(scope: Scope): void {
+      scope.setContext('Runtime', {
+        Name: nodeJs,
+        // global.process.version return the version of node
+        Version: global.process.version,
+      });
+    }
+
+    /**
+     * This function sets Additional Lambda Parameters which are displayed in Sentry Dashboard
+     * @param scope  - holds additional event information
+     * @hidden
+     */
+    function setAdditionalLambdaParameters(scope: Scope): void {
+      const remainingTimeInMillisecond: number = setupOnceContext.awsContext.getRemainingTimeInMillis() as number;
```

Comment:
`setupOnceContext` is only working because these functions are defined in the same scope as this variable, which is `setupOnce`. It'd be more "obvious" if we would pass `context` object instead. It's also easier to test this way.

Reply:
made this change in the new pull request.